scripts/pages/zoning/zoning.ts

```diff
@@ -1117,13 +1121,55 @@ class ZoneMenuHandler {
 		this.panels.teleportSection.visible = menu === RegionMenu.TELEPORT;
 	}
 
-	updateZones() {
+	drawZones() {
 		if (!this.mapZoneData) return;
+		if (!this.selectedZone) return;
 
-		// future: validation here
+		const renderRegions: ZoneEditorRegion[] = [];
 
-		this.mapZoneData.dataTimestamp = Date.now();
-		MomentumTimerAPI.SetActiveZoneDefs(this.mapZoneData);
+		const validity = this.isSelectionValid();
+		const regionIndex = validity.zone ? this.panels.regionSelect.GetSelected().GetAttributeInt('value', -1) : -1;
+
+		for (const [segmentNumber, segment] of this.selectedZone.track.zones.segments.entries()) {
+			if (segment.checkpoints.length > 0) {
+				for (const [checkpointNumber, checkpoint] of segment.checkpoints.entries()) {
+					for (const [_, region] of checkpoint.regions.entries()) {
```

Comment:
Doesn't look like you need the `.entries()` call here, since the index from `.entries()` is unused - just iterate over the original array!

`[foo, bar, baz].entries() -> [[0, foo], [1, bar], [2, baz]]` if you destructure that into `[_, item]` you're in effect just looping over the original array, with an unnecessary `.entries()` call and array access.

Reply:
good shout. Fixed

Reply:
Doesn't seem to have been pushed